aiven/client/connection_info/redis.py

```diff
@@ -0,0 +1,53 @@
+from ._utils import find_component, find_user, format_uri
+from .common import ConnectionInfoError
+
+
+class RedisConnectionInfo:
+    def __init__(self, host, port, username, db, password):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.db = db
+
+    @classmethod
+    def from_service(cls, service, *, route, usage, privatelink_connection_id, username, db):
+        if service["service_type"] != "redis":
+            raise ConnectionInfoError(
+                "Cannot format redis connection info for service type {service_type}".format_map(service)
+            )
+
+        info = find_component(
+            service["components"], route=route, usage=usage, privatelink_connection_id=privatelink_connection_id
+        )
+        host = info["host"]
+        port = info["port"]
+        if username == "default":
```

Comment:
What if I want to have `default` username?

Reply:
This is just a Redis specificity from Redis version < 6 not actually having users, and Redis >= 6 having implicit user called "default" when authenticating without a username.

Reply:
This is just a Redis specificity from Redis version < 6 not actually having users, and Redis >= 6 having implicit user called "default" when authenticating without a username.